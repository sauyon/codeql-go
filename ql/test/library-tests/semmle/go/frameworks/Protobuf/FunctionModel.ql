import go

from DataFlow::Node pred, DataFlow::Node succ
where
  exists(FunctionInput inp, FunctionOutput outp, DataFlow::CallNode call |
    exists(TaintTracking::FunctionModel f | call.getTarget() = f | f.hasTaintFlow(inp, outp))
    or
    exists(DataFlow::FunctionModel f | call.getTarget() = f | f.hasDataFlow(inp, outp))
  |
    pred = inp.getEntryNode(call) and succ = outp.getExitNode(call)
  )
select pred, succ
