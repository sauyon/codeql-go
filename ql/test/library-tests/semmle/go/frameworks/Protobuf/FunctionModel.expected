| proto.go:27:10:27:21 | reflectedXSS | proto.go:27:10:27:38 | call to GetDescription |
| proto.go:28:12:28:23 | reflectedXSS | proto.go:28:12:28:35 | call to GetAlerts |
| proto.go:30:35:30:46 | reflectedXSS | proto.go:30:2:30:47 | ... := ...[0] |
| proto.go:36:24:36:33 | serialized | proto.go:35:2:35:13 | definition of deserialized |
